dev/integration_tests/web_e2e_tests/test_driver/cache_width_cache_height_integration.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:ui' as ui;
+
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:integration_test/integration_test.dart';
+
+
+class LoadTestImageProvider extends ImageProvider<Object> {
+  LoadTestImageProvider(this.provider);
+
+  final ImageProvider provider;
+
+  ImageStreamCompleter testLoad(Object key, DecoderBufferCallback decode) {
+    return provider.loadBuffer(key, decode);
+  }
+
+  @override
+  Future<Object> obtainKey(ImageConfiguration configuration) {
+    throw UnimplementedError();
+  }
+
+  @override
+  ImageStreamCompleter loadBuffer(Object key, DecoderBufferCallback decode) {
+    throw UnimplementedError();
+  }
+}
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  testWidgets('Image.network uses cacheWidth and cacheHeight',
+          (WidgetTester tester) async {
+
+    const int expectedCacheHeight = 9;
+    const int expectedCacheWidth = 11;
+    await tester.pumpAndSettle();
+
+    final Image image = Image.network(
+      'assets/packages/flutter_gallery_assets/assets/icons/material/material.png',
+      cacheHeight: 9,
+      cacheWidth: 11,
+    );
+
+    bool called = false;
+
+    Future<ui.Codec> decode(ui.ImmutableBuffer buffer, {int? cacheWidth, int? cacheHeight, bool allowUpscaling = false}) {
+      expect(cacheHeight, expectedCacheHeight);
+      expect(cacheWidth, expectedCacheWidth);
+      expect(allowUpscaling, false);
+      called = true;
+      return PaintingBinding.instance.instantiateImageCodecFromBuffer(buffer, cacheWidth: cacheWidth, cacheHeight: cacheHeight, allowUpscaling: allowUpscaling);
+    }
+
+    final ImageProvider resizeImage = image.image;
+    expect(image.image, isA<ResizeImage>());
+    expect(called, false);
```

Comment:
This expectation doesn't seem to have an effect. AFAICT the `decode` function hasn't been used for anything at this point, so it couldn't have been called. Maybe it should go below `testProvider.testLoad`? (I'm actually not sure when it's expected to be called)

Reply:
Yea, on second thought, that check is not necessary. It might be fine to just check at the end if `called` has been flipped to true. It should be called with the `loadBuffer` call made by `testProvider.testLoad`.